client/src/components/prospects/Table.jsx

```diff
@@ -17,46 +19,32 @@ const useStyles = makeStyles({
   },
 });
 
-function createData(id, name, email) {
-  return { id, name, email };
-}
-
-// const rows = [
-// //   createData('1','Kim', 'kim@fakeemail.com'),
-// //   createData('2','Kim', 'kim@fakeemail.com'),
-// //   createData('3','Kim', 'kim@fakeemail.com'),
-// //   createData('4','Kim', 'kim@fakeemail.com'),
-// //   createData('5','Kim', 'kim@fakeemail.com')
-// // ];
-
 export default function ProspectsTable() {
   const classes = useStyles();
 
-  const [rows, setRows] = useState([
-    createData('1','Kim', 'kim@fakeemail.com')
-  ]);
+  const [rows, setRows] = useState([]);
+  const [needsUpdate, setNeedsUpdate] = useState(true);
 
   function fetchProspects() {
-    fetch("/api/prospects", {
-      method: "GET",
-      headers: {
-        "Content-Type": "application/json",
-        // TODO: This should be pulled from the local storage? or at least passed in.
-        "Authorization": "Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJleHAiOjE1OTYxNTQ0NDYsImlhdCI6MTU5NjE1MDg0Niwic3ViIjoxfQ.lZpkVZluz2ZLr6eljlwUabt-_b5N4p5C7rjexd30RKo"
-      }
-    }).then((res) => {
-      res.json().then((data) => {
-        console.log(data);
-        setRows(data.prospects);
-      });
+    Auth.getProspects().then((res) => {
+      console.log(res.data);
+      setRows(res.data.prospects);
     });
+
   }
 
+  // Gets called every time this component gets rendered (updated).
+  useEffect(() => {
+    // Do additional state check - if needsUpdate variable is true (initial render), then fetch prospect from API and
+    // set needsUpdate to false so that it only fetches once.
+    if (needsUpdate) {
+      fetchProspects();
+      setNeedsUpdate(false);
+    }
+  });
```

Comment:
you actually don't need needsUpdate variable, instead you can attain this by doing this:

```
useEffect(() => {
      fetchProspects();
  }, []);
```

The second parameter - look into the dependency array - this is more efficient as it for sure only runs once, and you don't need to keep running this useEffect whenever anything changes

Reply:
I still need to update this one. Thanks for the tip!